feat: implement 404 page

01001101CK opened the pull request:
<!-- 👋 Hi, thanks for sending a PR to squiggleconf.com! 💖
Please fill out all fields below and make sure each item is true and [x] checked.
Otherwise we may not be able to review your PR. -->

## PR Checklist

- [x] Steps in [CONTRIBUTING.md](https://github.com/SquiggleTools/SquiggleConf.com/blob/main/.github/CONTRIBUTING.md) were taken

## Overview
Add a 404 page with an ocean-themed error message

- Create 404.astro page with centered error message
- Use semantic color tokens for automatic light/dark mode support
- Include 'You've drifted into uncharted waters' message
- Add home page link button with proper styling
- Follow design system patterns without manual dark mode overrides

https://github.com/user-attachments/assets/ee2a845b-1c08-41f1-9167-7b11a0788de4

01001101CK committed:
add a 404 page

Signed-off-by: Yufa <112290188+01001101CK@users.noreply.github.com>

01001101CK committed:
1) reduce the number of extra colors in the color scheme. 2) remove specific .dark theme

01001101CK committed:
Merge branch 'SquiggleTools:main' into main

JoshuaKGoldberg committed:
Merge branch 'main' into feature/404-page

JoshuaKGoldberg commented on src/pages/404.astro:
```diff
@@ -0,0 +1,83 @@
+<PageLayout description="Page not found - SquiggleConf" title="404">
+	<div class="error-container">
+		<div class="error-box">
+			<div class="error-title">404: NOT_FOUND</div>
```
[Bug] Accessibility issue: each page should have at least an h1. https://www.w3.org/WAI/tutorials/page-structure/headings

```suggestion
			<h1 class="error-title">404: NOT_FOUND</h1>
```

JoshuaKGoldberg commented on src/pages/404.astro:
```diff
@@ -0,0 +1,83 @@
+		align-items: center;
+		justify-content: center;
+		min-height: 60vh;
+		padding: 2rem;
```
[Nit] I think the box is just a _little_ lower than the mid-point of the page. How about bumping it up a few rem to make it higher & more prominent?

```suggestion
    padding: 0 2rem 4rem;
```

JoshuaKGoldberg commented on src/pages/404.astro:
```diff
@@ -0,0 +1,83 @@
+			</div>
+		</div>
+
+		<a href="/" class="home-link"> Home Page </a>
```
Sorry, I'd meant to post this on the other PR but it's still pending. I think I forgot to submit it...

We have a component for this already. No need to reinvent new button styles.

```suggestion
		<Button as="a" href="/" variant="light">Home Page</Button>
```

JoshuaKGoldberg reviewed (changes requested):
This is looking great 🔥! I appreciate the using so many existing tokens and streamlining the code & styles. 

Just a few requests: the only real blocker from me is the `h1` one. How do they sound to you?

01001101CK committed:
Update src/pages/404.astro

Change div to h1

Co-authored-by: Josh Goldberg ✨ <git@joshuakgoldberg.com>

01001101CK committed:
Update src/pages/404.astro

making the box more prominent

Co-authored-by: Josh Goldberg ✨ <git@joshuakgoldberg.com>

01001101CK committed:
Update src/pages/404.astro

use existing button component

Co-authored-by: Josh Goldberg ✨ <git@joshuakgoldberg.com>

01001101CK commented:
> This is looking great 🔥! I appreciate the using so many existing tokens and streamlining the code & styles.
> 
> Just a few requests: the only real blocker from me is the `h1` one. How do they sound to you?

01001101CK committed:
- Fixed Node.js engine warning
- added pnpm.peerDependencyRules in package.json to allow @astrojs/markdown-remark v6 and astro v5 for packages that haven't updated their peer dependency ranges.
- Upgraded dependencies
